src/tasks/qa.py

```diff
@@ -172,3 +193,191 @@ def get_metrics(self, reset=False):
             self._score_tracker = collections.defaultdict(list)
 
         return {"ans_f1": ans_f1, "qst_f1": qst_f1, "em": em, "avg": (ans_f1 + em) / 2}
+
+
+@register_task("record", rel_path="ReCoRD/")
+class ReCoRDTask(Task):
+    """Reading Comprehension with commonsense Reasoning Dataset
+    See paper at https://sheng-z.github.io/ReCoRD-explorer """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        """ """
+        super().__init__(name, **kw)
+        self.val_metric = "%s_avg" % self.name
+        self.val_metric_decreases = False
+        self._score_tracker = collections.defaultdict(list)
+        self.max_seq_len = max_seq_len
+        self.files_by_split = {
+            "train": os.path.join(path, "train.jsonl"),
+            "val": os.path.join(path, "dev.jsonl"),
+            "test": os.path.join(path, "test.jsonl"),
+        }
+        # Load asnwers, used for computing metrics
+        self._load_answers()
+
+    def load_data(self):
+        # Data is exposed as iterable: no preloading
+        pass
+
+    def get_split_text(self, split: str):
+        """ Get split text as iterable of records.
+
+        Split should be one of "train", "val", or "test".
+        """
+        return self.load_data_for_path(self.files_by_split[split], split)
+
+    def load_data_for_path(self, path, split):
+        """ Load data """
+
+        def split_then_tokenize(sent):
+            """ Tokenize questions while preserving @placeholder token """
+            sent_parts = sent.split("@placeholder")
```

Comment:
What is this?

Reply:
ReCoRD's queries are Cloze style, so we want to tokenize while preserving the special placeholder token that indicates the masked out word.